geth/common/types.go

```diff
@@ -178,7 +178,7 @@ type TxQueueManager interface {
 type JailCell interface {
 	Set(string, interface{}) error
 	Get(string) (otto.Value, error)
-	Run(string) (otto.Value, error)
+	Run(interface{}) (otto.Value, error)
```

Comment:
This is needed because `(*JailCell).Run()` is a wrapper to `(*Otto).Run()` and otto's Run() accepts empty interface. It generally works with two possible types - `string` and `*otto.Script`.

Reply:
@tiabc I had some experience with ADRs and it didn't work well due to hard discoverability. When the new person looks at the code, the last place (s)he would look after explanations is adr/ folder in the root of the project, even if is aware of what ADR is. So the idea is nice, but I haven't seen it working well enough in practice.

Agree as for the comments.

Reply:
Ok, if you've had experience with it, I trust you because I, honestly, have had the same concerns before proposing it.